src/platform/switch/switch_input.c

```diff
@@ -261,58 +258,49 @@ void switch_handle_analog_sticks(void)
     float right_joy_dead_zone_squared = 10240.0*10240.0;
     float slope = 0.414214f; // tangent of 22.5 degrees for size of angular zones
 
-    if ((right_x * right_x + right_y * right_y) <= right_joy_dead_zone_squared) {
-        return;
-    }
+    int direction_states[ANALOG_MAX] = { 0, 0, 0, 0 };
 
-    int up = 0;
-    int down = 0;
-    int left = 0;
-    int right = 0;
-
-    if (right_y > 0 && right_x > 0) {
-        // upper right quadrant
-        if (right_y > slope * right_x) {
-            up = 1;
-        }
-        if (right_x > slope * right_y) {
-            right = 1;
-        }
-    } else if (right_y > 0 && right_x <= 0) {
-        // upper left quadrant
-        if (right_y > slope * (-right_x)) {
-            up = 1;
-        }
-        if ((-right_x) > slope * right_y) {
-            left = 1;
-        }
-    } else if (right_y <= 0 && right_x > 0) {
-        // lower right quadrant
-        if ((-right_y) > slope * right_x) {
-            down = 1;
-        }
-        if (right_x > slope * (-right_y)) {
-            right = 1;
-        }
-    } else if (right_y <= 0 && right_x <= 0) {
-        // lower left quadrant
-        if ((-right_y) > slope * (-right_x)) {
-            down = 1;
-        }
-        if ((-right_x) > slope * (-right_y)) {
-            left = 1;
+    if ((right_x * right_x + right_y * right_y) > right_joy_dead_zone_squared) {
+        if (right_y > 0 && right_x > 0) {
+            // upper right quadrant
+            if (right_y > slope *right_x) {
+                direction_states[ANALOG_UP] = 1;
+            }
+            if (right_x > slope *right_y) {
+                direction_states[ANALOG_RIGHT] = 1;
+            }
+        } else if (right_y > 0 && right_x <= 0) {
+            // upper left quadrant
+            if (right_y > slope *(-right_x)) {
```

Comment:
Unnecessary parentheses around `-right_x`
Comment also valid for the following `if` statements